data-prepper-plugins/parse-json-processor/src/main/java/org/opensearch/dataprepper/plugins/codec/json/JsonOutputCodec.java

```diff
@@ -30,69 +30,98 @@
 public class JsonOutputCodec implements OutputCodec {
     private final ObjectMapper objectMapper = new ObjectMapper();
     private static final String JSON = "json";
-    private static final JsonFactory factory = new JsonFactory();
+    private static final JsonFactory JSON_FACTORY = new JsonFactory();
     private final JsonOutputCodecConfig config;
-    private JsonGenerator generator;
-    private OutputCodecContext codecContext;
+    private JsonWriter deprecatedSupportWriter;
 
     @DataPrepperPluginConstructor
     public JsonOutputCodec(final JsonOutputCodecConfig config) {
         Objects.requireNonNull(config);
         this.config = config;
     }
 
+    private class JsonWriter implements Writer {
+        private final JsonGenerator generator;
+        private final OutputStream outputStream;
+        private final OutputCodecContext codecContext;
+
+        private JsonWriter(final OutputStream outputStream, final OutputCodecContext codecContext) throws IOException {
+            this.outputStream = outputStream;
+            this.codecContext = codecContext;
+            generator = JSON_FACTORY.createGenerator(outputStream, JsonEncoding.UTF8);
```

Comment:
getEstimatedSize() is called millions of times. Is it a good idea to create a new generator each time?